### PR TITLE
Add kubecost helm chart

### PR DESCRIPTION
**Pull request by stafot**
Issue: MM-36940

#### Summary
Adds kubecost helm-chart to have more granular costs allocation.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-36940

#### Release Note

```release-note
Adds kubecost helm-chart to have more granular costs allocation.

```

### CODE REVIEW

**Review by gabrieljackson — COMMENTED** [review]
Looks like we have some test failures to work through. Please re-request my review when those are good to go.

**Review by Szymongib — COMMENTED** [review]
I think failing tests come from here https://github.com/mattermost/mattermost-cloud/blob/master/internal/api/request_test.go#L16

Shoot me a DM in case of any problems with adjusting them.

File: internal/provisioner/kubecost.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	"github.com/mattermost/mattermost-cloud/internal/tools/aws"
+	"github.com/mattermost/mattermost-cloud/internal/tools/kops"
+	"github.com/mattermost/mattermost-cloud/model"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type kubecost struct {
+	awsClient      aws.AWS
+	environment    string
```
Comment (Szymongib) [inline internal/provisioner/kubecost.go:21]: Some fields in this struct seem not to be used, such as `awsClient`, `environment`, and `cluster`.  If those are not needed I would suggest removing them.

File: internal/provisioner/kubecost.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+func (k *kubecost) Migrate() error {
+	return nil
+}
+
+func (k *kubecost) NewHelmDeployment() *helmDeployment {
+	kubecostToken :=""
+	if len(os.Getenv(model.KubecostToken)) > 0 {
+		kubecostToken = "kubecostToken="+os.Getenv(model.KubecostToken)
+	}
```
Comment (Szymongib) [inline internal/provisioner/kubecost.go:121]: I think @stylianosrigas changes will make it easier to handle but for now, seems ok.

**Review by Szymongib — APPROVED** [review]
Thanks for adjusting tests!

Looks good from the Provisioner side, did not look too deeply into Helm values.

File: internal/provisioner/kubecost.go (hunk 2 above)
Comment (stafot) [reply internal/provisioner/kubecost.go:121]: You mean to handle it without environment variable?

File: model/cluster_utility.go [hunk 3]
```diff
@@ -126,7 +135,8 @@ func (h *UtilityGroupVersions) UnmarshalJSON(bytes []byte) error {
 		h.Teleport = &HelmUtilityVersion{Chart: oldUtilGrpVers.Teleport}
 		h.Pgbouncer = &HelmUtilityVersion{Chart: oldUtilGrpVers.Pgbouncer}
 		h.Stackrox = &HelmUtilityVersion{Chart: oldUtilGrpVers.Stackrox}
-		return nil
+		h.Kubecost = &HelmUtilityVersion{Chart: oldUtilGrpVers.Kubecost}		
+                 return nil
```
Comment (gabrieljackson) [inline model/cluster_utility.go:139]: nit: looks like there is an extra tab here.

**Review by gabrieljackson — APPROVED** [review]

**Review by spirosoik — APPROVED** [review]

**Pull request merged**